src/Symfony/Component/Config/Builder/ArrayShapeGenerator.php

```diff
@@ -0,0 +1,214 @@
+        } elseif (!$children) {
+            return "'array<array-key, mixed>'";
+        }
+
+        if ($node instanceof PrototypedArrayNode) {
+            $prototype = $node->getPrototype();
+            if ($prototype instanceof ArrayNode) {
+                return 'array<'.self::doGeneratePhpDoc($prototype, 1 + $nestingLevel).'>';
+            }
+
+            return 'array<'.self::handleNodeType($prototype).'>';
+        }
+
+        $shape .= "[\n";
+
+        $handleNode = function (NodeInterface $node, int $nestingLevel): string {
+            if ($node instanceof ArrayNode) {
+                return self::doGenerateJetBrainsArrayShape($node, 1 + $nestingLevel);
+            }
+
+            return "'".self::handleNodeType($node)."'";
+        };
+
+        /** @var BaseNode $child */
+        foreach ($children as $child) {
+            $shape .= \sprintf("%s'%s' => ", str_repeat(' ', 4 * $nestingLevel), $child->getName());
+
+            if ($child instanceof PrototypedArrayNode) {
+                $shape .= '['.$handleNode($child->getPrototype(), $nestingLevel).']';
+            } else {
+                $shape .= $handleNode($child, $nestingLevel);
+            }
+
+            $shape .= ','.self::generateInlinePhpDocForNode($child)."\n";
+        }
+
+        $shape .= str_repeat(' ', 4 * ($nestingLevel - 1)).']';
+
+        return $shape.(1 === $nestingLevel ? ')]' : '');
+    }
+
+    private static function dumpNodeKey(NodeInterface $node): string
+    {
+        return $node->getName().($node->isRequired() ? '' : '?');
+    }
+
+    private static function handleNumericNode(NumericNode $node): string
+    {
+        if ($node instanceof IntegerNode) {
+            $type = 'int<%s, %s>';
+        } elseif ($node instanceof FloatNode) {
+            $type = 'float<%s, %s>';
+        } else {
+            $type = 'int<%s, %s>|float<%1$s, %2$s>';
+        }
+
+        $min = $node->getMin() ?? 'min';
+        $max = $node->getMax() ?? 'max';
+
+        return \sprintf($type, $min, $max);
+    }
+
+    private static function prependPhpDocWithStar(string $shape): string
+    {
+        return str_replace("\n", "\n * ", $shape);
+    }
+
+    private static function generateInlinePhpDocForNode(BaseNode $node): string
+    {
+        $hasContent = false;
+        $comment = ' /* ';
+
+        if ($node->hasDefaultValue() || $node->getInfo() || $node->isDeprecated()) {
+            if ($node->isDeprecated()) {
+                $hasContent = true;
+                $comment .= 'Deprecated: '.$node->getDeprecation($node->getName(), $node->getPath())['message'].' ';
+            }
+
+            if ($info = $node->getInfo()) {
+                $hasContent = true;
+                $comment .= $info.' ';
+            }
+
+            if ($node->hasDefaultValue() && !\is_array($defaultValue = $node->getDefaultValue())) {
+                $hasContent = true;
+                $comment .= 'Default: '.json_encode($defaultValue).'. ';
```

Comment:
let's add  JSON_UNESCAPED_SLASHES, JSON_UNESCAPED_UNICODE, JSON_THROW_ON_ERROR and JSON_PRESERVE_ZERO_FRACTION flags